contracts/token/ERC1155/ERC1155.sol

```diff
@@ -0,0 +1,234 @@
+pragma solidity ^0.5.0;
+
+import "./IERC1155.sol";
+import "./IERC1155TokenReceiver.sol";
+import "../../math/SafeMath.sol";
+import "../../utils/Address.sol";
+import "../../introspection/ERC165.sol";
+
+/**
+ * @title Standard ERC1155 token
+ *
+ * @dev Implementation of the basic standard multi-token.
+ * See https://eips.ethereum.org/EIPS/eip-1155
+ * Originally based on code by Enjin: https://github.com/enjin/erc-1155
+ */
+contract ERC1155 is ERC165, IERC1155
+{
+    using SafeMath for uint256;
+    using Address for address;
+
+    // Mapping from token ID to owner balances
+    mapping (uint256 => mapping(address => uint256)) private _balances;
+
+    // Mapping from owner to operator approvals
+    mapping (address => mapping(address => bool)) private _operatorApprovals;
+
+    constructor()
+        public
+    {
+        _registerInterface(
```

Comment:
The calculation of the `interfaceId` in the `ERC165` uses the actual selectors to perform the operation. On the other hand, in the OZ libraries those are declared as a comment in the code (`ERC721`) and a hardcoded value is used. Although this way it is anti-error prone, it differs from our libraries. What other considerations/benefits could bring changing all our libraries to this way of getting the `interfaceId`?

Reply:
I think you've basically got it. It's really just to prevent errors/make stuff more explicit. I do want to remark that the hardcoded values actually do exist in the tests though.

What this benefit has to be weighed against is _unoptimized_ performance. I think Solidity does constant expression folding if the optimizer is enabled, so this shouldn't impact performance for contracts using the optimizer.

My take on this is that since there are many other places where performance gets traded for explicit readability in the code (e.g. the transformation of direct `_operatorApprovals` access to internal `isApprovedForAll` calls in this PR), this approach is more consistent with the _overall_ direction of OZ contracts.